### PR TITLE
Bump NFT contracts

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Key changes are:
1. `launchProjectFor` args
2. New way of getting a project's tiers (see `useNftRewards` hook)

**How to test:**
1. Launch project with NFTs
2. Check that they load 
3. Pay the project above the amount of an NFT 
4. Check your wallet to see if you received

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/hooks/v2/NftRewards.ts [hunk 1]
```diff
@@ -18,7 +19,10 @@ async function getRewardTierFromIPFS({
   contractNftRewardTier: ContractNftRewardTier
   index: number
 }): Promise<NftRewardTier> {
-  const url = contractNftRewardTier.tokenUri
+  const url = ipfsCidUrl(
+    decodeEncodedIPFSUri(contractNftRewardTier.encodedIPFSUri),
```
Comment (johnnyd-eth) [inline src/hooks/v2/NftRewards.ts:23]: IPFS CID now stored as a hash in the contracts to save gas

File: src/pages/v2/p/components/V2Dashboard.tsx [hunk 2]
```diff
@@ -165,7 +165,9 @@ export default function V2Dashboard({
     nftRewardsCIDs = CIDsOfNftRewardTiersResponse(nftRewardTiersResponse)
   }
   const { data: nftRewardTiers, isLoading: nftRewardTiersLoading } =
-    useNftRewards(nftRewardTiersResponse ?? [])
+    nftRewardsCIDs
+      ? useNftRewards(nftRewardTiersResponse ?? [])
+      : { data: undefined, isLoading: false }
```
Comment (johnnyd-eth) [inline src/pages/v2/p/components/V2Dashboard.tsx:170]: Gets rid of the `allTiers` error on non-NFT projects (closes #1690)

File: src/constants/contracts/rinkeby/juiceNftRewards.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const JBTiered721DelegateStoreAddress =
```
Comment (tomquirk) [inline src/constants/contracts/rinkeby/juiceNftRewards.ts:1]: **question** is this available in the npm package?

File: src/hooks/v2/transactor/LaunchProjectWithNftsTx.ts [hunk 4]
```diff
@@ -40,19 +41,24 @@ function getJBDeployTieredNFTRewardDataSourceData({
   ownerAddress: string
   directory: string
 }) {
-  const tiersArg: ContractNftRewardTier[] = Object.keys(nftRewards).map(cid => {
```
Comment (tomquirk) [inline src/hooks/v2/transactor/LaunchProjectWithNftsTx.ts:43]: might be worth keeping the type. wdyt?

File: src/pages/v2/p/components/V2Dashboard.tsx (hunk 2 above)
Comment (tomquirk) [reply src/pages/v2/p/components/V2Dashboard.tsx:170]: hmm. hooks shouldnt be called conditionally (React should actually be erroring in this case, or eslint).

I'm guessing the below should be working but isnt. Maybe the fix is there instead?

https://github.com/jbx-protocol/juice-interface/blob/main/src/hooks/v2/NftRewards.ts

File: src/utils/contracts/loadContract.ts [hunk 5]
```diff
@@ -124,10 +128,27 @@ const loadJBV1TokenPaymentTerminalContract = async (network: NetworkName) => {
   return contractJson
 }
 
-const loadJBNftRewardsDeployerContract = async () => {
+const loadJBTiered721DelegateProjectDeployerContract = async () => {
   const temporaryNftDeployerContractJson = {
```
Comment (tomquirk) [inline src/utils/contracts/loadContract.ts:132]: **question** i guess its not temporary anymore? maybe a rename

File: src/utils/ipfs.ts [hunk 6]
```diff
@@ -176,3 +179,18 @@ export function formatIpfsLink(ipfsLink: string) {
   const cid = ipfsLinkParts[ipfsLinkParts.length - 1]
   return `https://${DEFAULT_PINATA_GATEWAY}/ipfs/${cid}`
 }
+
+// How IPFS URI's are stored in the contracts to save storage (/gas)
+export function encodeIPFSUri(cid: string) {
```
Comment (tomquirk) [inline src/utils/ipfs.ts:184]: **praise** nice

**Review by tomquirk — CHANGES_REQUESTED** [review]

File: src/pages/v2/p/components/V2Dashboard.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/pages/v2/p/components/V2Dashboard.tsx:170]: Ah true for sure

File: src/hooks/v2/transactor/LaunchProjectWithNftsTx.ts (hunk 4 above)
Comment (johnnyd-eth) [reply src/hooks/v2/transactor/LaunchProjectWithNftsTx.ts:43]: It's a different type though. What we pass to the contract and what the contract gives back to us is slightly different now. I didn't think it was worth making a specific type for what we pass

File: src/constants/contracts/rinkeby/juiceNftRewards.ts (hunk 3 above)
Comment (johnnyd-eth) [reply src/constants/contracts/rinkeby/juiceNftRewards.ts]: Made `util` functions to load the latest deployed contract addresses from the NPM package. LMKWYT

File: package.json [hunk 7]
```diff
@@ -8,6 +8,7 @@
     "@jbx-protocol/contracts-v1": "2.0.0",
     "@jbx-protocol/contracts-v2-4.0.0": "npm:@jbx-protocol/contracts-v2@4.0.0",
     "@jbx-protocol/contracts-v2-latest": "npm:@jbx-protocol/contracts-v2@8.0.0",
+    "@jbx-protocol/juice-nft-rewards": "^4.0.0",
```
Comment (tomquirk) [inline package.json:11]:
```suggestion
    "@jbx-protocol/juice-nft-rewards": "4.0.0",
```

**Review by tomquirk — APPROVED** [review]

**Pull request merged**